src/shared/lib/routeUtils.ts

```diff
@@ -20,9 +20,9 @@ import {
 export type ExtractRouteParams<T> = string extends T
   ? Record<string, string>
   : T extends `${infer _Start}:${infer Param}/${infer Rest}`
-  ? { [k in Param | keyof ExtractRouteParams<Rest>]: string }
+  ? { [k in Param | keyof ExtractRouteParams<Rest>]?: string }
   : T extends `${infer _Start}:${infer Param}`
-  ? { [k in Param]: string }
+  ? { [k in Param]?: string }
```

Comment:
Updating react router also meant that the route parameter (e.g. how `/:token` becomes `{ token }`) is defined is also changed. React router now enforces these as an optional string, so we have to update the types here so they are an optional string.